backpack/core/derivatives/nll_base.py

```diff
@@ -0,0 +1,81 @@
+"""Partial derivative bases for NLL losses."""
+from math import sqrt
+from typing import List, Tuple
+
+from torch import Size, Tensor
+from torch.nn import CrossEntropyLoss
+
+from backpack.core.derivatives.basederivatives import BaseLossDerivatives
+from backpack.utils.subsampling import subsample
+
+
+class NLLLossDerivatives(BaseLossDerivatives):
+    """Partial derivative bases for NLL loss."""
+
+    def _sqrt_hessian_sampled(
+        self,
+        module: CrossEntropyLoss,
+        g_inp: Tuple[Tensor],
+        g_out: Tuple[Tensor],
+        mc_samples: int = 1,
+        subsampling: List[int] = None,
+    ) -> Tensor:
+        self._checks(module)
+
+        subsampled_input = subsample(module.input0, subsampling=subsampling)
+        dist = self._make_distribution(subsampled_input, mc_samples)
+
+        samples = dist.sample(sample_shape=Size([mc_samples]))
```

Comment:
I think this part does not fully reflect how we use the distribution and needs refactoring:
- `dist` is `p( y | subsampled_input )`
- We want to draw targets `y_tilde ~ dist`
- We use `y_tilde` to compute `loss_tilde = - log( p( y | subsampled_input ) )`
- We compute `grad(loss_tilde, subsampled_input)` which is proportional to the Hessian sqrt